File: 加密保护/VMProtect Ultimate v2.06/Examples/Free Pascal/Project1.pas
program Project1;

uses Windows, Messages, Sysutils, VMProtectSDK;

{$R Resource.res}

const
  IDD_DIALOG = 101;
  IDC_BUTTON_CHECKPASSWORD = 102;
  IDC_EDIT = 1000;

function MainHandler(hDlg: HWND; Msg: LongWord; wParam: WPARAM; lParam: LPARAM): LRESULT; stdcall;
var Buf: array [0..99] of Char;
begin
  VMProtectBegin('Test marker');
  Result:=0;
  case Msg of
   WM_INITDIALOG: Result:=1;
   WM_COMMAND:
    begin
     if LoWord(wParam) = IDCANCEL then
      begin
       EndDialog(hDlg, LoWord(wParam));
       Result:=1;
      end
     else
     if LoWord(wParam) = IDC_BUTTON_CHECKPASSWORD then
      begin
       GetDlgItemText(hDlg, IDC_EDIT, Buf, SizeOf(Buf));
       if (StrToIntDef(Buf,0) mod 17 = 13) then
        MessageBox(hDlg, VMProtectDecryptStringA('Correct password'), 'Information', MB_ICONINFORMATION + MB_OK)
       else
        begin
         MessageBox(hDlg, VMProtectDecryptStringA('Incorrect password'), 'Error', MB_ICONERROR + MB_OK);
	 SetFocus(GetDlgItem(hDlg, IDC_EDIT));
        end;
      end;
    end;
  end;
  VMProtectEnd;
end;

begin
  DialogBox(GetModuleHandle(nil),PChar(IDD_DIALOG),0,@MainHandler);
end.

File: 加密保护/VMProtect Professional v2.08/Examples/MinGW/Project1.cpp
#include <windows.h>
#include "Resource.h"
#include "VMProtectSDK.h"

LRESULT CALLBACK  MainHandler(HWND, UINT, WPARAM, LPARAM);
 
int APIENTRY WinMain(HINSTANCE hInstance,
                     HINSTANCE hPrevInstance,
                     LPSTR     lpCmdLine,
                     int       nCmdShow)
{
    DialogBox(GetModuleHandle(NULL), (LPCTSTR)IDD_DIALOG, NULL, (DLGPROC)MainHandler);

	return 0;
}

LRESULT CALLBACK MainHandler(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
{

    VMProtectBegin("Test marker");
    switch (message)
    {

    case WM_INITDIALOG:
        
        return TRUE;

    case WM_COMMAND:

        if (LOWORD(wParam) == IDCANCEL) 
        {
			EndDialog(hDlg, LOWORD(wParam));
            return TRUE;
        }

        if (LOWORD(wParam) == IDC_BUTTON_CHECKPASSWORD)
        {
 			char buf[100]; 

			GetDlgItemText(hDlg, IDC_EDIT, buf, 100);

            if (atoi (buf) % 17 == 13) 
  				MessageBox(hDlg, VMProtectDecryptStringA("Correct password"), "Information", MB_ICONINFORMATION + MB_OK);
			else
			{    
				MessageBox(hDlg, VMProtectDecryptStringA("Incorrect password"), "Error", MB_ICONERROR + MB_OK);
				SetFocus(GetDlgItem(hDlg, IDC_EDIT));
			}
		}
	}

	VMProtectEnd();
	return FALSE;
}
